File: spec/javascript/test_switch_cast.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('SwitchCast', 'src/switch_opcodes.bin', function(r) {
  assert.strictEqual(r.firstObj.value, 'foobar');
  assert.strictEqual(r.secondVal, 0x42);
  // unable to test "err_cast" here
});

File: spec/cpp_stl_11/test_switch_cast.cpp
#include <boost/test/unit_test.hpp>

#include <switch_cast.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_switch_cast) {
    std::ifstream ifs("src/switch_opcodes.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    switch_cast_t* r = new switch_cast_t(&ks);

    BOOST_CHECK_EQUAL(r->first_obj()->value(), "foobar");
    BOOST_CHECK_EQUAL(r->second_val(), 0x42);

    // Unable to test "err_cast": C++ static casts with pointer types
    // don't do any type checking and do not throw relevant
    // std::bad_cast.

    delete r;
}
